src/book.py

```diff
@@ -309,6 +308,49 @@ def snippet(self, length, on, groups=1, non_cont=False, with_replacement=True,
             self._text = snip._text
         else:
             return snip
+        
+    def translate_to_pos(self:BookText, inplace=False):
```

Comment:
I got an error when `self` was set to be a `BookText` object. I don't think `self` being set is really necessary and it throws an error so just change`self:BookText` to `self`.

Reply:
Fixed.